indy_node/server/domain_req_handler.py

```diff
@@ -180,31 +187,22 @@ def _validateNym(self, req: Request):
 
     def _validateNewNym(self, req: Request, op, originRole):
         role = op.get(ROLE)
-        r, msg = Authoriser.authorised(NYM,
-                                       originRole,
-                                       field=ROLE,
-                                       oldVal=None,
-                                       newVal=role)
-        if not r:
-            raise UnauthorizedClientRequest(
-                req.identifier,
-                req.reqId,
-                "{} cannot add {}".format(
-                    Roles.nameFromValue(originRole),
-                    Roles.nameFromValue(role))
-            )
+        self.write_req_validator.validate(req,
+                                          [AuthActionAdd(txn_type=NYM,
+                                                         field=ROLE,
+                                                         value=role)])
 
     def _validateExistingNym(self, req: Request, op, originRole, nymData):
         unauthorized = False
-        reason = None
         origin = req.identifier
         owner = self.idrCache.getOwnerFor(op[TARGET_NYM], isCommitted=False)
-        isOwner = origin == owner
+        is_owner = origin == owner
 
-        if not originRole == TRUSTEE and not isOwner:
+        if not originRole == TRUSTEE and not is_owner:
```

Comment:
This check needs to be part of common auth map rules and validator

Reply:
Done.